### PR TITLE
Polyhedron_demo: make the statistics exportable

### LINKED ISSUE #2035: Polyhedron_demo : enhancements for some plugins
## Issue Details
We should
- [x] test if the polyhedron is manifold during the selection-edition
- [x] add a widget specifying the current mode (selection, edition, ...) 
- [ ] the constraint deformation plane should be filled and not transparent (edit_polyhedron_plugin)
- [x] Make the statistics exportable

### PR DESCRIPTION
**Pull request by maxGimeno**
## Summary of Changes
Add a button in the statistics dialog to export the stats in a text file.

## Release Management
* Affected package(s):Polyhedron_demo
* Issue(s) solved (if any): fix #2035

### CODE REVIEW

File: Polyhedron/demo/Polyhedron/Statistics_on_item_dialog.ui [hunk 1]
```diff
@@ -63,6 +49,44 @@
      </item>
     </layout>
    </item>
+   <item>
+    <layout class="QHBoxLayout" name="horizontalLayout_2" stretch="0,1,0">
+     <item>
+      <widget class="QPushButton" name="exportButton">
+       <property name="text">
+        <string>Export</string>
```
Comment (lrineau) [inline Polyhedron/demo/Polyhedron/Statistics_on_item_dialog.ui]: The text should be "Export...", and even better: "&Export..." (that adds a shortcut <kbd>Alt+E</kbd>).

https://ux.stackexchange.com/q/9544

**Comment by lrineau** [comment]
There are conflicts, that seem easy to fix.

**Comment by maxGimeno** [comment]
Fixed

File: Polyhedron/demo/Polyhedron/MainWindow.cpp [hunk 2]
```diff
@@ -2032,3 +2034,53 @@ void MainWindow::set_facegraph_mode_adapter(bool is_polyhedron)
...
+      {
+        str.append(QString("  %1: \n")
+                   .arg(data.categories[j].first));
+        titles_limit+=data.categories[j].second;
+        for(;title<titles_limit; ++title)
+        {
+          str.append(QString("    %1: ").arg(data.titles.at(title)));
+          str.append(QString("%1\n").arg(sit->computeStats(title)));
+          title++;
+        }
```
Comment (lrineau) [inline Polyhedron/demo/Polyhedron/MainWindow.cpp]: @maxGimeno: There is a warning about this loop in recent testsuites:
https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-4.12-Ic-59/Polyhedron_Demo/TestReport_lrineau_ArchLinux-clang-CXX14.gz
```
/home/cgal_tester/build/src/cmake/platforms/ArchLinux-clang-CXX14/test/Polyhedron_Demo/MainWindow.cpp:2066:11: warning: variable 'title' is incremented both in the loop header and in the loop body [-Wfor-loop-analysis]
          title++;
          ^
/home/cgal_tester/build/src/cmake/platforms/ArchLinux-clang-CXX14/test/Polyhedron_Demo/MainWindow.cpp:2062:36: note: incremented here
        for(;title<titles_limit; ++title)
                                   ^
```

Is that wanted? Anyway, even if that is wanted, I agree with the warning that it is confusing and it should be replaced by a single `title+=2`.

Comment (maxGimeno) [reply Polyhedron/demo/Polyhedron/MainWindow.cpp]: It was absolutely not wanted. Fixed in #2532.

**Pull request merged**